src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -787,6 +787,49 @@ void QgsWFSProvider::reloadProviderData()
   mShared->invalidateCache();
 }
 
+QDomElement QgsWFSProvider::geometryElement( const QgsGeometry &geometry, QDomDocument &transactionDoc )
+{
+  QDomElement gmlElem;
+
+  // Determine axis orientation and gml version
+  bool applyAxisInversion;
+  QgsOgcUtils::GMLVersion gmlVersion;
+
+  if ( mShared->mWFSVersion.startsWith( QLatin1String( "1.1" ) ) )
+  {
+    // WFS 1.1.0 uses preferably GML 3, but ESRI mapserver in 2020 doesn't like it so we stick to GML2
+    if ( ! mShared->mServerPrefersCoordinatesForTransactions_1_1 )
+    {
+      gmlVersion = QgsOgcUtils::GML_3_1_0;
+    }
+    else
+    {
+      gmlVersion = QgsOgcUtils::GML_2_1_2;
+    }
+    // For servers like Geomedia and QGIS Server that advertise EPSG:XXXX in capabilities even in WFS 1.1 or 2.0
+    // cpabilities useEPSGColumnFormat is set.
+    // We follow GeoServer convention here which is to treat EPSG:4326 as lon/lat
+    applyAxisInversion = ( crs().hasAxisInverted() && ! mShared->mURI.ignoreAxisOrientation() && ! mShared->mCaps.useEPSGColumnFormat )
+                         || mShared->mURI.invertAxisOrientation();
+  }
+  else // 1.0
+  {
+    gmlVersion = QgsOgcUtils::GML_2_1_2;
+    applyAxisInversion = mShared->mURI.invertAxisOrientation();
+  }
+
+  gmlElem = QgsOgcUtils::geometryToGML(
+              geometry,
+              transactionDoc,
+              gmlVersion,
+              mShared->srsName(),
```

Comment:
I guess it could be a bit funky if you use WFS 2.0 on the reading side that returns a urn::bla:EPSG:4326 and on the transaction side you use that when outputing the srsName in the geometry with WFS-T 1.0. Servers might have issues to interpret that correctly. They will probably assume lon, lat order because of WFTS-T 1.0, but the urn::bla:EPSG:4326 that could suggest lat, lon order.  Wouldn't we also have issues with the feature ids ? There's no guarantee that the gml:id of WFS 2.0 might be the fid of WFS-T 1.0 ...

Reply:
I used srsName here when testing geoserver: it sends `urn:...` as a default SRS and that means latitude/longitude, but if we use `EPSG:4326` it assumes longitude/latitude. 

According to geoserver docs, 
-    WFS 1.0.0: Provides geographic coordinates in east/north and may not be trusted to respect the EPSG definition axis order.
-   WFS 1.1.0: Respects the axis order defined by the EPSG definition.
-   WFS 2.0.0: Respects the axis order defined by the EPSG definition.

Anyway, besides to implement 2.0 what would you recommend? I can't find a better solution to fit all situations.

Reply:
> I used srsName here when testing geoserver: it sends urn:... as a default SRS and that means latitude/longitude, but if we use EPSG:4326 it assumes longitude/latitude.

So currently, let's say with GeoServer with WFS 2.0 that returns a srsName="urn::...EPSG:4326", for geometries we emit in transactions as WFS-T 1.0, do they have srsName="urn::...EPSG:4326" and lat,lon order ?  Or srsName="EPSG:4326" and lon,lat order ?

> Anyway, besides to implement 2.0 what would you recommend?

At least for GeoServer and other well behaved servers, using WFS-T 1.1 with WFS 2.0 would seem to me to be a safer solution since they should have the same axis order behaviour and gml:id

Reply:
If GeoServer WFS 2.0 sends `urn::...EPSG:4326` (or  `urn:x-ogc:def:crs:EPSG:4326`)  in capabilities and we emit a transaction WFS-T 1.0 it will use srsName `urn::...EPSG:4326` lat/lon (inverted axis).

But if a server sends `EPSG:4326` (like QGIS does) we follow GeoServer convention which is to treat `EPSG:4326` as lon/lat.

Reply:
> we emit a transaction WFS-T 1.0 it will use srsName `urn::...EPSG:4326` lat/lon (inverted axis).

OK, that's what I understood as well from the code. This is probably a "non-vetted" combination. Anyway I just wanted to point of potential issues I could think of. I let you judge of the best / least-worse conclusion :-)

Reply:
Honestly I'm a bit lost into all the combinations of implementations/specification-flaws and workarounds that we can find in the wild, I hope this is a good compromise. 

We could even add a separate option (hack) for inverted axis in transactions or to choose the WFS-T version for transactions but it looks to me that the list of options (and potentially incompatible ones) could be really overwhelming.

Anyway, thank you for the review and all the help!

Do you feel like we can merge (and backport) this PR?

Reply:
> Do you feel like we can merge (and backport) this PR?

The only situation where this changes behaviour underneath is when WFS 1.1 is used for the reading side: now, we'll switch to WFS-T 1.1, which is very reasonable.
Yes, we can probably treat that as a bug fix.
I'm wondering about the new UI string added: what's our policy regarding this ? Are there translation updates in the stable branch ?

Reply:
I don't know if we have real policy, probably better to avoid changes but a new little string in advanced options is probably fine.

Reply:
OK go ahead :-)